src/tmpfiles/tmpfiles.c

```diff
@@ -554,6 +554,58 @@ static nsec_t load_statx_timestamp_nsec(const struct statx_timestamp *ts) {
         return ts->tv_sec * NSEC_PER_SEC + ts->tv_nsec;
 }
 
+static int opendir_and_stat(
+                const char *path,
+                DIR **ret,
+                nsec_t *ret_atime_nsec,
+                nsec_t *ret_mtime_nsec,
+                dev_t *ret_rootdev_major,
+                dev_t *ret_rootdev_minor,
```

Comment:
I am not sure it's worth having four separate params here, if you could also just pass the statx structure here?

Reply:
also, i am not sure about using `dev_t` as type here... the statx has grown the field sizes substantially on this, for a reason...